app/collections/mutations/changeEditorRole.ts

```diff
@@ -17,14 +17,31 @@ export default resolver.pipe(
       },
     })
 
+    // Set up the counter for the number of owners and admin users
     let ownerAdmins = 0
+    let owners = 0
+
+    // Count the total number of existing owners and admin members
     for (const element of oldEditorship?.collection.editors!) {
       if (element.role === "OWNER" || element.role === "ADMIN") {
         ownerAdmins += 1
       }
+      if (element.role === "OWNER") owners += 1
     }
+    // Predict the number of owners and admins after the requested change
+    if (role === "OWNER" || role === "ADMIN") ownerAdmins += 1
```

Comment:
```suggestion
    if ((role === "OWNER" || role === "ADMIN") && oldEditorship.role === "USER") ownerAdmins += 1
```

I am trying to follow this logic, but struggling without access to a test environment. It seems like we might need to include a check that the editor is in fact not already an `OWNER || ADMIN`, such that we don't overcount the `ownerAdmins`?

For example, we might end in the situation where somebody tries to change to USER as last OWNER - gets an error, and then changes back to OWNER. Here we might be double counting the amount of `ownerAdmins` and not hitting the condition of `ownerAdmins === 1`.

What do you think @nsunami?

Reply:
Thanks for the catch, Chris! 

While testing this, I found a problem, which is related to the situation that you describe. 

1. Set up a collection with at least another user:
![image](https://user-images.githubusercontent.com/17035406/193329015-83473de1-c070-4f5a-9c13-0988f7dc1af7.png)

2. Change the other person's role as an owner
![image](https://user-images.githubusercontent.com/17035406/193329105-6c9dc674-8d7f-49cb-8ec4-dc74414701b7.png)

3. Set my role as an admin
![image](https://user-images.githubusercontent.com/17035406/193329174-394c854b-869f-4725-93ff-a1e32d9ea383.png)

4. Set the other person's role as a user
![image](https://user-images.githubusercontent.com/17035406/193329266-0bbb9546-b00a-424b-b344-aca093d2cbcc.png)

5. Now we have a collection without an owner, which should not be the case.

I will need to think about this.